src/views/GovernanceReviewTeam/Actions/IssueLifecycleId.tsx

```diff
@@ -38,24 +48,34 @@ const IssueLifecycleId = () => {
   const onSubmit = (values: SubmitLifecycleIdForm) => {
     const {
       feedback,
-      expirationDateMonth,
-      expirationDateDay,
-      expirationDateYear,
+      expirationDateMonth = '',
+      expirationDateDay = '',
+      expirationDateYear = '',
       nextSteps,
       scope,
       lifecycleId
     } = values;
-    const lcidPayload = {
-      lcidExpiresAt: `${expirationDateYear}-${expirationDateMonth}-${expirationDateDay}`,
-      decisionNextSteps: nextSteps,
-      lcidScope: scope,
+    const expiresAt = DateTime.utc(
+      parseInt(expirationDateYear, RADIX),
+      parseInt(expirationDateMonth, RADIX),
+      parseInt(expirationDateDay, RADIX)
+    );
+    const input = {
+      intakeId: systemId,
+      expiresAt: expiresAt.toISO(),
+      nextSteps,
+      scope,
       lcid: lifecycleId,
       feedback
     };
-    const payload = { id: systemId, lcidPayload };
-    dispatch(issueLifecycleIdForSystemIntake(payload));
 
-    history.push(`/governance-review-team/${systemId}/intake-request`);
+    mutate({
+      variables: { input }
+    }).then(response => {
```

Comment:
I'm curious how this works. What's the difference between having `response.errors` versus having `.catch()`?

Reply:
@christopherhuii the mutation response is outlined here: https://www.apollographql.com/docs/react/data/mutations/#result. Is there another way you would have expected to see this code written?

Reply:
Not specifically. I'm trying to understand when `.catch()` would be used versus checking for `errors` in `response`.

I would have expected this to be written as the following, but I don't know how the errors in catch differ from `response.errors`.
```
mutate({
  variables: { input }
}).then(() => {
  history.push(`/governance-review-team/${systemId}/decision`);
}).then(errors => {
   // do something with errors
});
```

Reply:
@christopherhuii Got it! The useMutation hook also gives us access to the mutationResponse in the component overall, so we can use that to pull the errors. In this specific case, I kept getting rerouted even when I had errors in the call, so I wanted to prevent that. Here's a code block that shows the errors:
```
 {mutationResult.error && (
  <ErrorAlert heading="Error issuing lifecycle id">
    <ErrorAlertMessage
      message={mutationResult.error.message}
      errorKey="systemIntake"
    />
  </ErrorAlert>
)}
```

I saw it in action when my lifecycle ID wasn't formatted properly. It wasn't a _nice_ error message, but it's better than our current state of ignoring it altogether.